docs/proposals/image-updater-first-class-support.md

```diff
@@ -0,0 +1,310 @@
+
+### Detailed examples
+
+
+At present, users of Image Updater must add a host of specific annotations to their applications in order to configure automatic image updates on them. As the user base of the Image Updater project has grown, the use cases and requirements have also gotten more complex and demanding. As these features have been implemented and delivered to the users, the ways to use annotations to leverage these features have become more complicated, involving somewhat convoluted mechanisms to express references to resources like secrets, and setting up templates for branch names. Using annotations also opens up the possibility of misconfigurations by means of usage of "random" annotation names (containing typos) which make it cumbersome to troubleshoot issues.
+Aside from providing strict type validation, converting these fields to first class fields in the CR would simplify the expression and usage of these fields and make them more intuitive for users to understand and leverage. 
+
+Example of proposed additions to Application CR spec and status to accomodate image update configuration is provided below:
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Application
+metadata:
+  ...
+
+spec:
+...
+
+  image:
+    # updates contains all the fields to configure exactly which images should be considered for automatic updates and how they should be updated
+    updates:
+      # ImageList contains list of image specific configuration for each image that is to be considered as a candidate for updation 
+      imageList:
+        - image: quay.io/some/image 
+
+          # (optional) Specify Semver constraint for allowed tags 
+          constraint: 1.17.x
+
+          # (optional) specify list of regex/wildcard patterns to further filter which tags should be considered for updates
+          allowTags: 
+            matchType: regex/wildcard
+            matchList:
+            - <pattern 1>
+            - <pattern 2>
+
+          # (optional) specify list of regex/wildcard patterns to dictate which tags should be ignored for updates
+          ignoreTags: 
+            matchType: regex/wildcard
+            matchList:
+            - <pattern 1>
+            - <pattern 2>
+
+          # (optional) force updates to images that don't appear in the status of an Argo CD application. Default is false   
+          forceUpdate: true/false
+
+          # specify strategy to be followed when determining which images have qualifying updates 
+          updateStrategy: semver/digest/lexical/most-recently-built
```

Comment:
should `most-recently-built` match the existing value `latest`?

Reply:
`latest` will be deprecated in Image Updater by the next iteration and is to be replaced by `most-recently-build`.